62.CAD/Hatch.cpp

```diff
@@ -0,0 +1,782 @@
+				//	curveBox.curveMin[0].x, curveBox.curveMin[0].y,
+				//	curveBox.curveMin[1].x, curveBox.curveMin[1].y,
+				//	curveBox.curveMin[2].x, curveBox.curveMin[2].y,
+				//	curveBox.curveMax[0].x, curveBox.curveMax[0].y,
+				//	curveBox.curveMax[1].x, curveBox.curveMax[1].y,
+				//	curveBox.curveMax[2].x, curveBox.curveMax[2].y
+				//).c_str());
+
+                hatchBoxes.push_back(curveBox);
+            }
+
+            // advance and trim all of the beziers in the candidate set
+            auto oit = activeCandidates.begin();
+            for (auto iit = activeCandidates.begin(); iit != activeCandidates.end(); iit++)
+            {
+                const double evalAtMajor = iit->originalBezier->evaluateBezier(iit->t_end)[major];
+
+				auto origBez = iit->originalBezier;
+				//std::cout << "Candidate: (" << origBez->p[0].x << ", " << origBez->p[0].y << "),"
+				//	"(" << origBez->p[1].x << ", " << origBez->p[1].y << ")," <<
+				//	"(" << origBez->p[2].x << ", " << origBez->p[2].y << ") " <<
+				//	"Evaluated at major: " << evalAtMajor;
+                // if we scrolled past the end of the segment, remove it
+                // (basically, we memcpy everything after something is different
+                // and we skip on the memcpy for any items that are also different)
+                // (this is supposedly a pattern with input/output operators)
+                if (newMajor < evalAtMajor)
+                {
+                    const double new_t_start = iit->originalBezier->intersectOrtho(newMajor, major);
+
+					//std::cout << " new_t_start = " << new_t_start << " minor at t_start = " << iit->originalBezier->evaluateBezier(new_t_start)[minor];
+                    // little optimization (don't memcpy anything before something was removed)
+                    if (oit != iit)
+                        *oit = *iit;
+                    oit->t_start = new_t_start;
+                    oit++;
+                }
+				//std::cout << "\n";
+            }
+            // trim
+            const auto newSize = std::distance(activeCandidates.begin(), oit);
+            activeCandidates.resize(newSize);
+
+			//std::cout << "New candidate size: " << newSize << "\n";
+        }
+		// If we had a start event, we need to add the candidate
+		if (addStartSegmentToCandidates)
+		{
+			addToCandidateSet(nextStartEvent);
+		}
+		// We'll need to sort if we had a start event and added to the candidate set
+		// or if we have advanced our candidate set
+		if (addStartSegmentToCandidates || newMajor > lastMajor)
+		{
+			std::sort(activeCandidates.begin(), activeCandidates.end(), candidateComparator);
+			if (newMajor > lastMajor)
+				lastMajor = newMajor;
+		}
+		step++;
+    }
+	debugStep = debugStep - step;
+}
+
+
+// returns two possible values of t in the second curve where the curves intersect
+std::array<double, 4> Hatch::QuadraticBezier::linePossibleIntersections(const QuadraticBezier& second) const
```

Comment:
I don't want to review this function :)))
I trust you implemented correctly based on the desmos + book chapter, and the tests you've been doing